system/cache/util/EventURLFacade.cfc

```diff
@@ -35,36 +35,72 @@ component accessors="true" {
 
 	/**
 	 * Build a unique hash from an incoming request context
+     * Note: the 'event' key is always ignored from the request collection
 	 *
 	 * @event A request context object
+     * @targetContext The targeted request context object
+     * @eventDictionary The event metadata containing cache annotations
 	 */
-	string function getUniqueHash( required event ){
-		var rcTarget = arguments.event
-			.getCollection()
-			.filter( function( key, value ){
-				// Remove event, not needed for hashing purposes
-				return ( key != "event" );
-			} );
-
-		// systemOutput( "=====> uniquehash-rcTarget: #variables.jTreeMap.init( rcTarget ).toString()#", true );
-		// systemOutput( "=====> uniquehash-rcTargetHash: #hash( variables.jTreeMap.init( rcTarget ).toString() )#", true );
-
-		var targetMixer = {
-			// Get the original incoming context hash
-			"incomingHash" : hash( variables.jTreeMap.init( rcTarget ).toString() ),
-			// Multi-Host support
-			"cgihost"      : buildAppLink()
-		};
-
-		// Incorporate Routed Structs
-		structAppend(
-			targetMixer,
-			arguments.event.getRoutedStruct(),
-			true
-		);
-
-		// Return unique identifier
-		return hash( targetmixer.toString() );
+	string function getUniqueHash( 
+        required event,
+        required struct eventDictionary
+    ){
+        
+        // Assign the RC struct and filter out the "event" key, which is not needed for cache keys
+        var rcTarget = arguments.event.getCollection().filter( ( key, value ) => { 
+            return key != "event";
+        } );
+        
+        // Apply cache key filtering based on annotations
+        // We apply them in the following order:
+        // 1. Custom filter closure (if provided)
+        // 2. Include specific keys (if `cacheInclude` is not "*")
+        // 3. Exclude specific keys (if `cacheExclude` is provided and not empty)
+        
+        // Use custom filter closure stored in dictionary
+        if ( len( arguments.eventDictionary?.cacheFilter ) ) {
```

Comment:
Also note that the ?. is not necessary anymore, since we are seeding all metadata with it.
So the valuation is `!isSimpleValue( )`

Reply:
I think my latest PR addresses this.